samples/openapi3/client/petstore/dart2/petstore_client_lib/lib/auth/http_basic_auth.dart

```diff
@@ -11,14 +11,19 @@
 part of openapi.api;
 
 class HttpBasicAuth implements Authentication {
-  HttpBasicAuth({this.username, this.password});
+  HttpBasicAuth({this.username = '', this.password = ''});
 
-  String? username;
-  String? password;
+  String username;
+  String password;
 
   @override
   void applyToParams(List<QueryParam> queryParams, Map<String, String> headerParams) {
-    final credentials = '${username ?? ''}:${password ?? ''}';
-    headerParams['Authorization'] = 'Basic ${base64.encode(utf8.encode(credentials))}';
+    username = username.trim();
```

Comment:
why the trim?

Reply:
if a username of password ends with ` ` space user won't be able to login as wrong password will be sent to a server, user generated API has not way to fix it without modifying generated code, which then would be overriden in the next generation.

Reply:
Got it, I'll fix then and upload new code.

Reply:
:heavy_check_mark: